imports/plugins/core/versions/server/migrations/8_update_registry_provides_to_array.js

```diff
@@ -39,7 +39,8 @@ Migrations.add({
         // of the array
         const updatedRegistry = pkg.registry.map((entry) => {
           if (Array.isArray(entry.provides)) {
-            entry.provides = entry.provides[0];
+            const [provides] = entry.provides;
+            entry.provides = provides;
```

Comment:
This is one example of it actually being more confusing to read, and more lines, with destructuring. Maybe we just want to disable the rule for this line, and any similar lines? Thoughts @spencern ?

Reply:
I'm actually more concerned with the `entry.provides[0]` to start with.

grabbing just the first item from the entry.provides array makes it seem like we need a way to set a default on the `provides` item, or that we should be selecting the default instead of the 0th index if we already have a way to identify a default.

If the 0th index is intentional and okay for now, then I'm okay with disabling.

In general, our lint rules should help code be more readable and/or more performant. If it's not doing either, that's a good candidate for using disable.

Reply:
I think it's better to revert this to previous state (and avoid confusion in reading it). And disable the rule for this line.
@spencern your concern about the [0] is actually noted in the comments above this line. There is a description about why it was written this way